recipes/kokkos-kernels/meta.yaml

```
@@ -0,0 +1,64 @@
+{% set name = "kokkos-kernels" %}
+{% set version = "3.7.01" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/kokkos/kokkos-kernels/archive/refs/tags/{{ version }}.tar.gz
+  sha256: b2060f5894bdaf7f7d4793b90444fac260460cfa80595afcbcb955518864b446
+
+build:
+  skip: true  # [win]
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('kokkos-kernels') }}
+
+requirements:
+  build:
+    - cmake >=3.21.1,<4
+    - {{ compiler('cxx') }}
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+    - ninja
+  host:
+    - kokkos
+  run:
+    - kokkos
+
+test:
+  source_files:
+    - example/batched_solve
+  commands:
+    - $CXX example/batched_solve/team_GMRES.cpp -o team_GMRES -I$PREFIX/include -L$PREFIX/lib -lkokkoscontainers -lkokkoscore -lkokkoskernels -fopenmp
```

We may need to use C++17. It fails on macOS.
```suggestion
    - $CXX example/batched_solve/team_GMRES.cpp -o team_GMRES -std=c++17 -I$PREFIX/include -L$PREFIX/lib -lkokkoscontainers -lkokkoscore -lkokkoskernels -fopenmp
```